### PR TITLE
48 - one-click script for development environment setup for Linux

### PR DESCRIPTION
**Pull request by F15Ky**
Configuration of postgres not implemented yet since I am not sure if we are going to be using postgres. 

I am not sure of the best way to install with nix with a config file since the user may already have a config file and there appears to be no standard way of doing it.

I Tested this script on Ubuntu linux.

### CODE REVIEW

**Comment by BstevensG4G** [comment]
In my experience there are issues installing most relational databases in my personal work environment "WSL2" on Win10. Mainly starting the db service itself. I'm still not familiar enough with Linux on having the db server service start automatically like in Windows, although it seems easy enough to start manually in linux

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/bash
+
+install_nix () {
+    if ! command -v nix-env &> /dev/null
```
Comment (razvan-flavius-panda) [inline install.sh:4]: Please create a function for `command -v ... &> /dev/null`

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        curl -L https://nixos.org/nix/install | sh
+        . /home/ben/.nix-profile/etc/profile.d/nix.sh
+    fi
+    if [ ! -f ~/.config/nixpkgs/config.nix ];
+    then
+        mkdir -p ~/.config/nixpkgs/
+        echo -n "with import <nixpkgs> {};
+
+        let
+            all-hies = import (fetchTarball \"https://github.com/infinisil/all-hies/tarball/master\") {};
```
Comment (razvan-flavius-panda) [inline install.sh:15]: Remove this as it is not needed

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            unstable = import <nixpkgs> { inherit config; };
+        in
+          {
+            allowUnfree = true;
+            allowBroken = true;
```
Comment (razvan-flavius-panda) [inline install.sh:20]: Remove, not needed

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    then
+        mkdir -p ~/.config/nixpkgs/
+        echo -n "with import <nixpkgs> {};
+
+        let
+            all-hies = import (fetchTarball \"https://github.com/infinisil/all-hies/tarball/master\") {};
+            unstable = import <nixpkgs> { inherit config; };
+        in
+          {
+            allowUnfree = true;
```
Comment (razvan-flavius-panda) [inline install.sh:19]: Remove, not needed

**Review by razvan-flavius-panda — CHANGES_REQUESTED** [review]
All good, just a few minor issues

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    done
+fi
+
+if ! command -v cachix &> /dev/null
+then
+    nix-env -iA cachix -f https://cachix.org/api/v1/install
+fi
+cachix use everythingmanager
+nix-shell
+cabal update
```
Comment (BstevensG4G) [inline install.sh:82]: Luv luv luv this!!

File: install.sh (hunk 2 above)
Comment (F15Ky) [reply install.sh:15]: I get errors when I remove this saying it cant find the derivation for ghc8104

**Review by razvan-flavius-panda — APPROVED** [review]

File: install.sh (hunk 2 above)
Comment (razvan-flavius-panda) [reply install.sh:19]: @F15Ky HIE is obsoleted by HLS so this should be removed, try to figure out a solution to this please

**Review by razvan-flavius-panda — CHANGES_REQUESTED** [review]
See my comment about HIE removal

**Review by razvan-flavius-panda — APPROVED** [review]

**Pull request merged**